secret-manager/pkg/backend/cache/metrics/metrics.go

```diff
@@ -30,12 +37,13 @@ func init() {
 func registerMetrics(reg prometheus.Registerer) {
 	registerOnce.Do(func() {
 		reg.MustRegister(cacheAccess)
+		reg.MustRegister(CacheSize)
 	})
 }
 
 // RecordCacheHit increments the counter for a successful cache hit
-func RecordCacheHit() {
-	cacheAccess.WithLabelValues("hit", "").Inc()
+func RecordCacheHit(reason string) {
+	cacheAccess.WithLabelValues("hit", reason).Inc()
```

Comment:
See my previous comments: Add a new label: `method` to the metrics, which can be `get`, `set` (or alternative `read`, `write`)